pkg/intermediate/aggregate.go

```diff
@@ -515,45 +516,43 @@ func (a *AggregationProcess) aggregateRecords(incomingRecord, existingRecord ent
 			existingIeWithValue, index, _ := existingRecord.GetInfoElementWithValue(element)
 			// Update the corresponding element in existing record.
 			if !isDelta {
-				if existingIeWithValue.Value.(uint64) < ieWithValue.Value.(uint64) {
-					existingIeWithValue.Value = ieWithValue.Value
-				}
-			} else {
-				// We are simply adding the delta stats now. We expect delta stats to be
-				// reset after sending the record from flowKeyMap in aggregation process.
-				// Delta stats from source and destination nodes are added, so we will have
-				// two times the stats approximately.
-				// For delta stats, it is better to use source and destination specific
-				// stats.
-				existingIeWithValue.Value = existingIeWithValue.Value.(uint64) + ieWithValue.Value.(uint64)
-			}
-			if err := existingRecord.SetInfoElementWithValue(index, *existingIeWithValue); err != nil {
-				return err
+				existingIeWithValue.Value = util.MaxUint64(existingIeWithValue.Value.(uint64), ieWithValue.Value.(uint64))
 			}
 			// Update the corresponding source element in antreaStatsElement list.
 			if fillSrcStats {
-				existingIeWithValue, index, _ = existingRecord.GetInfoElementWithValue(antreaSourceStatsElements[i])
+				srcExistingIeWithValue, index, _ := existingRecord.GetInfoElementWithValue(antreaSourceStatsElements[i])
 				if !isDelta {
-					existingIeWithValue.Value = ieWithValue.Value
+					srcExistingIeWithValue.Value = ieWithValue.Value
 				} else {
-					existingIeWithValue.Value = existingIeWithValue.Value.(uint64) + ieWithValue.Value.(uint64)
+					srcExistingIeWithValue.Value = srcExistingIeWithValue.Value.(uint64) + ieWithValue.Value.(uint64)
+					// delta count takes the maximum of the current delta count and
+					// the updated delta count from source node
+					existingIeWithValue.Value = util.MaxUint64(existingIeWithValue.Value.(uint64), srcExistingIeWithValue.Value.(uint64))
```

Comment:
how about filling up src and dst stat elements first and then fill the common stat elements? I think that will improve the readability of the code and we dont have to update the common stat elements at two different places.

Reply:
make sense to me. Thanks Srikar!